bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaSilentRollerShutterHandler.java

```diff
@@ -45,25 +45,37 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             return;
         } else {
             String cmd = getTahomaCommand(command.toString());
-            if (COMMAND_MY.equals(cmd)) {
-                sendCommand(COMMAND_MY);
-            } else if (COMMAND_STOP.equals(cmd)) {
-                String executionId = getCurrentExecutions();
-                if (executionId != null) {
-                    // Check if the roller shutter is moving and STOP is sent => STOP it
-                    cancelExecution(executionId);
-                } else {
+            switch (cmd) {
+                case COMMAND_MY:
                     sendCommand(COMMAND_MY);
-                }
-            } else {
-                if (CONTROL_SILENT.equals(channelUID.getId()) && COMMAND_SET_CLOSURE.equals(cmd)) {
-                    // move the roller shutter to the specific position at low speed
-                    String param = "[" + command.toString() + ", \"lowspeed\"]";
-                    sendCommand(COMMAND_SET_CLOSURESPEED, param);
-                } else {
-                    String param = COMMAND_SET_CLOSURE.equals(cmd) ? "[" + toInteger(command) + "]" : "[]";
-                    sendCommand(cmd, param);
-                }
+                    break;
+                case COMMAND_STOP:
+                    String executionId = getCurrentExecutions();
+                    if (executionId != null) {
+                        // Check if the roller shutter is moving and STOP is sent => STOP it
+                        cancelExecution(executionId);
+                    } else {
+                        sendCommand(COMMAND_MY);
+                    }
+                    break;
```

Comment:
You can do similar fall through shenanigans here as well by moving the COMMAND_MY case around.
```suggestion
                case COMMAND_STOP:
                    String executionId = getCurrentExecutions();
                    if (executionId != null) {
                        // Check if the roller shutter is moving and STOP is sent => STOP it
                        cancelExecution(executionId);
                        break;
                    }
                    //fall through
                case COMMAND_MY:
                    sendCommand(COMMAND_MY);
                    break;
```

Reply:
Great suggestion. thanks